### PR TITLE
First stab at GitHub Actions

### PR DESCRIPTION
**Pull request by iMacTia**
## Description
First configuration of a CI workflow using GitHub Actions that runs Rubocop, Yard-Junk and Rspec.

## Todos
List any remaining work that needs to be done, i.e:
- [x] Rubocop
- [x] Yard-Junk
- [x] RSpec
- [x] Rubygems deployment
- [x] CodeClimate

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Ruby 2.6
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
+    - name: Rubocop
+      run: |
+        gem install rubocop rubocop-junit_formatter rubocop-performance --no-document
+        rubocop --require rubocop/formatter/junit_formatter \
```
Comment (olleolleolle) [inline .github/workflows/ruby.yml]:
```suggestion
        rubocop \
```

**Comment by iMacTia** [comment]
@olleolleolle I was wondering what that was, I thought it was related to CodeClimate 😅 
Thanks for clarifying!

**Comment by iMacTia** [comment]
Looks good now but I had to explicitly install `libcurl`, which should not be necessary.
I've opened an issue [here](https://github.com/actions/virtual-environments/issues/37) to shed some light on it

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Set up Ruby 2.6
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
+    - name: Rubocop
+      run: |
+        gem install rubocop rubocop-performance --no-document
+        rubocop --require rubocop-performance \
+                --format progress \
+                --out ~/test-results/linting/rubocop.xml
```
Comment (olleolleolle) [inline .github/workflows/ci.yml]: We can stop creating the output as a file. We are not using that.

So, remove the `--out ...` option.

File: README.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 [![Gem Version](https://badge.fury.io/rb/faraday.svg)](https://rubygems.org/gems/faraday)
 [![CircleCI](https://circleci.com/gh/lostisland/faraday/tree/master.svg?style=svg)](https://circleci.com/gh/lostisland/faraday/tree/master)
```
Comment (olleolleolle) [inline README.md]: We can drop the CircleCI badge in the same go.

```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,7 +15,6 @@ jobs:
         gem install rubocop rubocop-performance --no-document
         rubocop --require rubocop-performance \
                 --format progress \
-                --out ~/test-results/linting/rubocop.xml
```
Comment (olleolleolle) [inline .github/workflows/ci.yml]: Oh, the line continuation line on the line before should also be removed.

**Comment by olleolleolle** [comment]
The feedback runs are so fast.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (olleolleolle) [reply .github/workflows/ci.yml]: It still _worked_, though.

**Comment by iMacTia** [comment]
Yeah it looks much faster than any external CI 😮!
When I opened the PR, it already started building as well so it was a better of seconds before I could see the result!

File: .github/workflows/ci.yml (hunk 4 above)
Comment (olleolleolle) [reply .github/workflows/ci.yml]: Perhaps we can reduce all the way down to `rubocop --require rubocop-performance`

**Pull request merged**